environment.yml

```diff
@@ -30,3 +30,7 @@ dependencies:
   # Development
   - jupyterlab
   - nbsphinx
+
+  # Pip
+  - pip:
+    - flatten-dict
```

Comment:
@janosg might have thoughts on whether this or [pybaum](https://github.com/OpenSourceEconomics/pybaum) is more useful for the job. pybaum is being used extensively in the ecosystem around dags, so might be better to stick to one implementation.

Reply:
I did not find a function in pybaum that transforms a nested dict like `{"a": {"b": 1}}` into a flattened dict like `{"a__b": 1}`. Instead pybaum's `tree_flatten` seems to return a list containing only the leaves (`[1]`).

Reply:
The main difference between the approaches in pybaum and flatten_dict is that unflattening in `flatten-dict` works with the flat dictionary alone. All information about the tree structure is preserved in the keys. In pybaum, unflattening requires knowledge about the tree structure (the so called `treedef`, that is returned from `tree_flatten`). This is necessary because we usually flatten to an unlabeled list, where no structure is preserved. 

I think this is an argument for using `flatten-dict` in gettsim.